nav2_bt_navigator/behavior_trees/navigate_w_replanning.xml

```diff
@@ -4,14 +4,24 @@
 
 <root main_tree_to_execute="MainTree">
   <BehaviorTree ID="MainTree">
-    <Sequence name="root">
-      <RateController hz="1.0">
-        <Fallback>
-          <GoalReached/>
-          <ComputePathToPose goal="{goal}" planner_property="GridBased"/>
-        </Fallback>
-      </RateController>
-      <FollowPath path="{path}"/>
-    </Sequence>
+    <FirstCompletion name="root">
+      <Repeat num_cycles="-1">
+        <RateController hz="1.0">
+          <Fallback>
+            <GoalReached/>
+            <Sequence name="ComputePathSequence">
+              <ComputePathToPose goal="{goal}" path="{path}" planner_property="GridBased"/>
+              <SetBlackboard value="1" output_key="first_planning_complete" />
+            </Sequence>
+          </Fallback>
+        </RateController>
+      </Repeat>
+      <BlackboardCheckInt name="Wait for first path" 
```

Comment:
What is `BlackboardCheckInt`? I dont see it defined or described? It looks like checking if some variable is some value. Is there a reason the `first_planning_complete` check isn't baked into the controller node or that the controller node cannot be called before a plan has been created? That seems like architectural way of doing it since under no situation would `FollowPath` be called without a path to follow

Reply:
`BlackboardCheckInt` is (an undocumented) part of the BehaviorTree library.

> Is there a reason the first_planning_complete check isn't baked into the controller node

That could be done. I had an idea for what I was calling a `ParallelSequence`. It would be like a `ReactiveSequence` but would allow multiple subtrees to be RUNNING simultaneously.

However, I'm currently working on the assumption that BehaviorTree control nodes should be generally useful and not navigation specific. I could find no precedent for my `ParallelSequence` node concept and the very name I came up with seemed an oxymoron, which didn't bode well. So, I went with this  `FirstCompletion` node type which I had seen discussed in threads about `Parallel` nodes.

To keep it generally useful, I moved the specific logic of our use case into the tree using the `SetBlackboard` and `BlackboardCheckInt` nodes.

That's my rationale. I'm open to discussion.